### PR TITLE
Remove extra output in bash completion

### LINKED ISSUE #583: Extra output and errors during shell completion
**What steps did you take:**
There is extra output in the shell completion choices and in some cases is even causes errors during shell completion.

**What happened:**

For bash, notice the error in this particular case (some other bash completions do work):
```
bash
bash-5.1$ source <(kapp completion bash)
bash-5.1$ kapp help <TAB>
bash: ShellCompDirectiveNoFileComp

Succeeded: syntax error in expression (error token is "Succeeded")
```
For zsh, notice the extra output at the end (`:4`, `Succeeded`, etc):
```
$ source <(kapp completion zsh)
$ compdef _kapp kapp
$ kapp <TAB>
app-change                                                     -- App change (garbage-collect, list)
app-group                                                      -- app-group will deploy/delete an application for each subdirectory within a
completion                                                     -- Output shell completion code for the specified shell (bash, zsh or fish)
config-map                                                     -- Config map (list)
delete                                                         -- Delete app
deploy                                                         -- Deploy app
deploy-config                                                  -- Show default deploy config
help                                                           -- Help about any command
inspect                                                        -- Inspect app
label                                                          -- Print specified app label
list                                                           -- List all apps in a namespace
logs                                                           -- Print app's Pod logs
rename                                                         -- Rename app
service-account                                                -- Service account (list)
tools                                                          -- Tools (diff, inspect, list-labels)
version                                                        -- Print client version
:4                                                             Succeeded
Completion ended with directive: ShellCompDirectiveNoFileComp
```
Also happens for fish shell.

**What did you expect:**

Shell completion to work as expected, without error or extra output.

**Anything else you would like to add:**

Shell completion uses the hidden `__complete` command provided by the Cobra project to obtain the shell completion choices.  
1. `kapp` should not print `Succeeded` for that command
2. output printed to `stderr` by the `__complete` command should not be redirected by `kapp` to `stdout` (that is the problem for the string output `Completion ended with directive: ShellCompDirectiveNoFileComp`)

This command is added by Cobra when `rootCmd.Execute()` is called, which makes it a little trickier to adjust the behaviour for it.

**Environment:**

- kapp version (use `kapp --version`): 0.52.0
- OS (e.g. from `/etc/os-release`): MacOS
- Kubernetes version (use `kubectl version`): N/A

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

### PR DESCRIPTION
**Pull request by praveenrewar**
#### What this PR does / why we need it:
- Do not print Succeeded for help and hidden __complete command
- Use SetOut instead of deprecated SetOutput for setting output for help command.

#### Which issue(s) this PR fixes:

Fixes #583 

#### Does this PR introduce a user-facing change?

```release-note

```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

File: cmd/kapp/kapp.go [hunk 1]
```diff
@@ -45,6 +47,22 @@ func nonExitingMain() error {
...
+		var cmdName string // first "non-flag" arguments
+		for _, arg := range cmdPathPieces {
+			if !strings.HasPrefix(arg, "-") {
+				cmdName = arg
+				break
+			}
+		}
+		switch cmdName {
+		case "help", cobra.ShellCompRequestCmd, cobra.ShellCompNoDescRequestCmd:
+			// do not print Succeeded
```
Comment (kumaritanushree) [inline cmd/kapp/kapp.go]: Suggestion: do not print Succeeded -> Do not print...

Comment (100mik) [reply cmd/kapp/kapp.go]: +1 to this ^

**Comment by cppforlife** [comment]
why do we need to special case this? wouldnt all of these commands (case "help", cobra.ShellCompRequestCmd, cobra.ShellCompNoDescRequestCmd) be running in non-tty env, so `Succeeded` shouldnt be showing up already?

**Comment by praveenrewar** [comment]
> why do we need to special case this? wouldnt all of these commands (case "help", cobra.ShellCompRequestCmd, cobra.ShellCompNoDescRequestCmd) be running in non-tty env

These commands are a bit special, quoting from the issue itself:

> This command is added by Cobra when rootCmd.Execute() is called, which makes it a little trickier to adjust the behaviour for it.

So, right now when we [decide](https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/cmd/kapp.go#L141) which commands should have the default tty behaviour and which should not have it, these commands are not even added to the command tree 😞

**Comment by rohitagg2020** [comment]
LGTM

**Review by rohitagg2020 — APPROVED** [review]

**Review by 100mik — APPROVED** [review]
The cobrautil path looks much cleaner!
LGTM 🚀

**Pull request merged**